mantidproject/mantid
# Save files multiple directories eng diff UI

**Description of work.**
Save calibration and focused data to two directories if RB number is specified and ROI not texture. See issue for more details.

**To test:**
More detailed instructions can be found here
https://developer.mantidproject.org/Testing/EngineeringDiffraction/EngineeringDiffractionTestGuide.html
(1) Open the ENgDiff UI
(2) type something the the RB num box at top - e.g. `SaveFilesPR`
(3) Run a calibration for both banks e.g. using run 193749 (both banks is the default when cropping is not checked)
(4) Check the calibration files (.prm) are saved to two directories where SaveLocation is the directory displayed at the bottom of the UI window (can be change din the settings)
1. `SaveLocation/Calibration`
2. `SaveLocation/User/SaveFilesPR/Focus`

(5) Delete the files and check that when no RB num is specified it would just save the files to the first directory.
(6) Check the Crop Calibration box and select texture as the region of interest and re-run a calibration
(7) Check the files are saved to directory 1 (when no RB num specified)
(8) Write something in the RB num box and re-run the texture calibration
(9) Check the files are saved to directory 2


Fixes #33039

*This does not require release notes* because **it restores the behaviour of the previous major version for all ROI other than texture (which was added this release)**




---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
